### PR TITLE
change envoy UPDATE_BRANCH to 1.31 for 1.23

### PR DESCRIPTION
**Pull request by thedebugger**

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by dhawton** [comment]
Istio 1.23 should be Envoy 1.31, right? 1.22 was 1.30.. and we typically +1 both. Right now the main branch is 1.31?

**Review by dhawton — CHANGES_REQUESTED** [review]
Leave on main until they cut 1.31.

File: prow/cluster/jobs/istio/proxy/istio.proxy.release-1.23.gen.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ periodics:
       - --labels=auto-merge
       - --modifier=update_envoy_dep
       - --token-env
-      - --cmd=UPDATE_BRANCH=main scripts/update_envoy.sh
+      - --cmd=UPDATE_BRANCH=release/v1.30 scripts/update_envoy.sh
```
Comment (mikemorris) [inline prow/cluster/jobs/istio/proxy/istio.proxy.release-1.23.gen.yaml]:
```suggestion
      - --cmd=UPDATE_BRANCH=release/v1.31 scripts/update_envoy.sh
```

File: prow/config/jobs/proxy-1.23.yaml [hunk 2]
```diff
@@ -1306,7 +1306,7 @@ jobs:
   - --labels=auto-merge
   - --modifier=update_envoy_dep
   - --token-env
-  - --cmd=UPDATE_BRANCH=main scripts/update_envoy.sh
+  - --cmd=UPDATE_BRANCH=release/1.30 scripts/update_envoy.sh
```
Comment (mikemorris) [inline prow/config/jobs/proxy-1.23.yaml]:
```suggestion
  - --cmd=UPDATE_BRANCH=release/1.31 scripts/update_envoy.sh
```

**Review by mikemorris — CHANGES_REQUESTED** [review]
We'll stay on Envoy `main` branch until 1.31 is published. Suggesting changes and blocking until https://github.com/envoyproxy/envoy/pull/35188 is complete.

**Review by mikemorris — APPROVED** [review]

**Comment by zirain** [comment]
https://github.com/envoyproxy/envoy/releases/tag/v1.31.0 is out, good to go!

**Review by kfaseela — APPROVED** [review]

**Review by dhawton — APPROVED** [review]

**Pull request merged**